### ISSUE TITLE
IE crashes when retrieving http-only session cookies on IE Driver 3.14

### ISSUE THREAD

**Issue by savsav** [SeleniumHQ/selenium#6261]
## Meta -
OS:  Windows 7
Selenium Version:  IEDriver v3.14 (32-bit)
Browser:  Internet Explorer v.11.0.9600.13035 (32-bit)

## Expected Behavior -
Cookies are returned by the API.

## Actual Behavior -
The GET request to '**/session/[session-id]/cookie**' is stuck on executing. WebdriverIO was used as test framework but this works on other browsers and a simple curl request reproduces the issue as well (curl -X GET http://127.0.0.1:5555/session/[session-id]/cookie).
Reproduces on sites with http-only & session cookies (e.g. https://www.next.co.uk/ - see the example below).

<details><summary>Show IE Driver Log</summary>
<p>T 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\webdriver-server\server.cc(242) Process request with: URI: /session/</p>02df7de6-5112-4eb7-8518-47bea7b97262/cookie HTTP verb: GET
body: {}
<p>T 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\webdriver-server\server.cc(331) Entering Server::DispatchCommand</p>
<p>T 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\webdriver-server\server.cc(661) Entering Server::LookupCommand</p>
D 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\webdriver-server\server.cc(340) Command: GET /session/02df7de6-5112-4eb7-8518-47bea7b97262/cookie {}
<p>T 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\webdriver-server\server.cc(456) Entering Server::LookupSession</p>
<p>T 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\iedriver\iesession.cpp(191) Entering IESession::ExecuteCommand</p>
<p>T 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(131) Entering IECommandExecutor::OnSetCommand</p>
<p>T 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\webdriver-server\command.cc(31) Entering Command::Deserialize</p>
D 2018-08-08 16:38:55:852 c:\projects\webdriver\cpp\webdriver-server\command.cc(36) Raw JSON command: { "name" : "getAllCookies", "locator" : { "sessionid" : "02df7de6-5112-4eb7-8518-47bea7b97262" }, "parameters" : {} }
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(142) Entering IECommandExecutor::OnExecCommand</p>
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\iesession.cpp(212) Beginning wait for response length to be not zero</p>
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(666) Entering IECommandExecutor::DispatchCommand</p>
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(890) Entering IECommandExecutor::GetCurrentBrowser</p>
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(896) Entering IECommandExecutor::GetManagedBrowser</p>
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(829) Entering IECommandExecutor::IsAlertActive</p>
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\browser.cpp(714) Entering Browser::GetActiveDialogWindowHandle</p>
<p>T 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\browser.cpp(218) Entering Browser::GetContentWindowHandle</p>
D 2018-08-08 16:38:55:853 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(843) No alert handle is found
<p>T 2018-08-08 16:38:55:854 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(890) Entering IECommandExecutor::GetCurrentBrowser</p>
<p>T 2018-08-08 16:38:55:854 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(896) Entering IECommandExecutor::GetManagedBrowser</p>
<p>T 2018-08-08 16:38:55:854 c:\projects\webdriver\cpp\iedriver\documenthost.cpp(74) Entering DocumentHost::GetCurrentUrl</p>
<p>T 2018-08-08 16:38:55:854 c:\projects\webdriver\cpp\iedriver\browser.cpp(138) Entering Browser::GetDocument</p>
I 2018-08-08 16:38:55:854 c:\projects\webdriver\cpp\iedriver\browser.cpp(142) No child frame focus. Focus is on top-level frame
<p>T 2018-08-08 16:38:55:875 c:\projects\webdriver\cpp\iedriver\browser.cpp(627) Entering Browser::GetDocumentFromWindow</p>
<p>T 2018-08-08 16:38:55:882 c:\projects\webdriver\cpp\iedriver\cookiemanager.cpp(95) Entering CookieManager::GetCookies</p>
<p>T 2018-08-08 16:38:55:882 c:\projects\webdriver\cpp\iedriver\hookprocessor.cpp(105) Entering HookProcessor::Initialize</p>
<p>T 2018-08-08 16:38:55:882 c:\projects\webdriver\cpp\iedriver\hookprocessor.cpp(201) Entering HookProcessor::CreateReturnPipe</p>
D 2018-08-08 16:38:55:882 c:\projects\webdriver\cpp\iedriver\hookprocessor.cpp(236) Created named pipe \\.\pipe\IEDriverPipe6488
<p>T 2018-08-08 16:38:55:882 c:\projects\webdriver\cpp\iedriver\hookprocessor.cpp(164) Entering HookProcessor::InstallWindowsHook</p>
<p>T 2018-08-08 16:38:55:883 c:\projects\webdriver\cpp\iedriver\cookiemanager.cpp(409) Entering CookieManager::SendGetCookieMessage</p>
<p>T 2018-08-08 16:38:55:883 c:\projects\webdriver\cpp\iedriver\hookprocessor.cpp(258) Entering HookProcessor::PushData</p>
<p>T 2018-08-08 16:38:55:888 c:\projects\webdriver\cpp\iedriver\hookprocessor.cpp(284) Entering HookProcessor::PullData</p>
D 2018-08-08 16:38:55:888 c:\projects\webdriver\cpp\iedriver\hookprocessor.cpp(290) Waiting for connection from browser via named pipe
<p>T 2018-08-08 16:38:55:888 c:\projects\webdriver\cpp\iedriver\cookiemanager.cpp(438) Entering CookieManager::ThreadProc</p>
</details>

## Steps to reproduce -
Just try to retrieve the cookies from a site (make sure there are HTTP-only session cookies as well - see the example):

```javascript
browser.url('https://www.next.co.uk/');
let cookies = browser.cookie(); // gets all cookies - GET request to /session/[session-id]/cookie
```

**Comment by jimevans** [comment]
Fixed in 20bdf47. Thank you for the issue report.

**Comment by jimevans** [comment]
For those keeping score at home, one actual issue had nothing to do with HttpOnly or session cookies. The Windows API used to retrieve the cookie values was returning a `NULL` pointer for the value property of the cookie, and the IE driver was not properly checking for that before manipulating the value. If this change does not fix the issue, then there's something else happening in a legacy code path.
